src/OfficeService.Data.Provider.MsSql.Ef/Migrations/20220405150400_AddWorkspaces.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+
+using LT.DigitalOffice.OfficeService.Data.Provider.MsSql.Ef;
+using LT.DigitalOffice.OfficeService.Models.Db;
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace LT.DigitalOffice.OfficeService.Data.Migrations
+{
+  [DbContext(typeof(OfficeServiceDbContext))]
+  [Migration("20220405150400_AddWorkspaces")]
+  public class AddWorkspaces : Migration
+  {
+    #region Create tables
+
+    private void CreateWorkspacesTable(MigrationBuilder migrationBuilder)
+    {
+      migrationBuilder.CreateTable(
+        name: DbWorkspace.TableName,
+        columns: table => new
+        {
+          Id = table.Column<Guid>(nullable: false),
+          ParentId = table.Column<Guid?>(nullable: true),
+          Name = table.Column<string>(nullable: false),
+          WorkspaceTypeId = table.Column<Guid>(nullable: false),
+          Description = table.Column<string>(nullable: true),
+          IsBookable = table.Column<bool>(nullable: false),
+          CreatedBy = table.Column<Guid>(nullable: false),
+          CreatedAtUtc = table.Column<DateTime>(nullable: false),
+          ModifiedBy = table.Column<Guid>(nullable: true),
+          ModifiedAtUtc = table.Column<DateTime>(nullable: true),
+          IsActive = table.Column<bool>(nullable: false)
+        },
+        constraints: table =>
+        {
+          table.PrimaryKey("PK_Workspaces", w => w.Id);
+        });
+    }
+
+    private void CreateWorkspacesTypesTable(MigrationBuilder migrationBuilder)
+    {
+      migrationBuilder.CreateTable(
+        name: DbWorkspaceType.TableName,
+        columns: table => new
+        {
+          Id = table.Column<Guid>(nullable: false),
+          Name = table.Column<string>(nullable: false, maxLength: 100),
+          Description = table.Column<string>(nullable: true),
+          StartTime = table.Column<string>(nullable: true),
+          EndTime = table.Column<string>(nullable: true),
+          BookingRule = table.Column<int>(nullable: false),
+          CreatedBy = table.Column<Guid>(nullable: false),
+          CreatedAtUtc = table.Column<DateTime>(nullable: false),
+          ModifiedBy = table.Column<Guid>(nullable: true),
+          ModifiedAtUtc = table.Column<DateTime>(nullable: true),
+          IsActive = table.Column<bool>(nullable: false)
+        },
+        constraints: table =>
+        {
+          table.PrimaryKey("PK_WorkspacesTypes", w => w.Id);
+        });
+
+      migrationBuilder.AddUniqueConstraint(
+        name: "UX_Workspaces_Name_Unique",
```

Comment:
You add UniqueConstraint to WorkspaseType, why do you name as Workspase?
"UX_Workspaces_Name_Unique" => "UC_WorkspacesTypes_Name"

Reply:
my bad